www/waterfall_view/src/module/main.module.js

```diff
@@ -635,14 +640,89 @@ var WaterfallController = (function() {
             });
         }
 
-        renderNewData() {
+        toggleTag(tag) {
+            if (!this.$scope.tags_filter.includes(tag)) {
+                this.$scope.tags_filter.push(tag);
+            } else {
+                this.$scope.tags_filter = _.remove(this.$scope.tags_filter, function(currentTag) {
+                    return currentTag != tag;
+                });
+            }
+            this.$location.search("tags", this.$scope.tags_filter);
+            this.renderNewData(this.$scope.tags_filter);
+        }
+
+        isTagToggled(tag) {
```

Comment:
nit: toggle is an action, not a state.

Then, you method do not return a boolean, but returns a class.

I would call this method isTagFiltered, and would push the class logic into the template